addons/l10n_hr_edi/models/account_move.py

```diff
@@ -0,0 +1,197 @@
+    # Additional fields
+    l10n_hr_edi_addendum_id = fields.One2many(comodel_name='l10n_hr.edi_addendum', inverse_name='move_id', string='HR EDI Addendum', copy=False)
+    l10n_hr_invoice_sending_time = fields.Datetime(related='l10n_hr_edi_addendum_id.l10n_hr_invoice_sending_time')
+    # EDI and fiscalization-specific fields
+    l10n_hr_business_document_status = fields.Selection(related='l10n_hr_edi_addendum_id.l10n_hr_business_document_status')
+    l10n_hr_business_status_reason = fields.Char(related='l10n_hr_edi_addendum_id.l10n_hr_business_status_reason')
+    l10n_hr_fiscalization_number = fields.Char(related='l10n_hr_edi_addendum_id.l10n_hr_fiscalization_number')
+    l10n_hr_fiscalization_status = fields.Selection(related='l10n_hr_edi_addendum_id.l10n_hr_fiscalization_status')
+    l10n_hr_fiscalization_error = fields.Char(related='l10n_hr_edi_addendum_id.l10n_hr_fiscalization_error')
+    l10n_hr_fiscalization_request = fields.Char(related='l10n_hr_edi_addendum_id.l10n_hr_fiscalization_request')
+    l10n_hr_fiscalization_channel_type = fields.Selection(related='l10n_hr_edi_addendum_id.l10n_hr_fiscalization_channel_type')
+    # MojEracun integration fields
+    l10n_hr_mer_document_id = fields.Char(related='l10n_hr_edi_addendum_id.l10n_hr_mer_document_id')
+    l10n_hr_mer_document_status = fields.Selection(related='l10n_hr_edi_addendum_id.l10n_hr_mer_document_status')
+
+    @api.constrains('move_type', 'l10n_hr_process_type')
+    def _check_l10n_hr_process_type(self):
+        for record in self:
+            if record.country_code == 'HR' and (record.l10n_hr_process_type == 'P9') == (record.move_type != 'out_refund'):
+                raise ValidationError(self.env._('Business Process Type P9 can only be used with credit notes and vice versa.'))
+
+    @api.depends('l10n_hr_fiscalization_status')
+    def _compute_show_reset_to_draft_button(self):
+        # EXTENDS 'account'
+        super()._compute_show_reset_to_draft_button()
+        for move in self:
+            if move.l10n_hr_fiscalization_status:
+                move.show_reset_to_draft_button = False
+
+    @api.depends('move_type', 'l10n_hr_process_type')
+    def _compute_l10n_hr_process_type(self):
+        for move in self:
+            if not move.l10n_hr_process_type:
+                if move.move_type == 'out_refund':
+                    move.l10n_hr_process_type = 'P9'
+                elif move.move_type == 'out_invoice':
+                    move.l10n_hr_process_type = 'P1'
+
+    def _get_l10n_hr_fiscalization_numer(self, name):
+        """
+        Extract the fiscal numbering triple (ex. 1/1/1) from the document name.
+        Only applies for Croatian sales invoices/credit notes. Expected name
+        pattern is produced by the overridden `sequence.mixin` logic.
+        """
+        parts = name.split('-')
+        if len(parts) >= 3:
+            return str(int(parts[2].split('/')[0])) + '/' + '/'.join(parts[2].split('/')[1:])
```

Comment:
I feel that even though not necessarily as powerful, a regex here would help future readers understand more clearly what the expected string format is, and what we are extracting from it.

e.g. something like
`r'(?P<year>\d+)-(?P<month>\d+)-(?P<seq>\d+)/(?P<branch_number>\d+)/(?P<other_number>\d+)'`